crates/node/builder/src/components/mod.rs

```diff
@@ -43,7 +43,7 @@ pub trait NodeComponents<T: FullNodeTypes>: Clone + Unpin + Send + Sync + 'stati
     type Pool: TransactionPool<Transaction: PoolTransaction<Consensus = TxTy<T::Types>>> + Unpin;
 
     /// The node's EVM configuration, defining settings for the Ethereum Virtual Machine.
-    type Evm: ConfigureEvm<Header = HeaderTy<T::Types>, Transaction = TxTy<T::Types>>;
+    type Evm: ConfigureEvmFor<<T::Types as NodeTypes>::Primitives>;
```

Comment:
we could also introduce an alias for this `PrimitivesFor` for example

Reply:
yep